apps/src/JsDebuggerUi.js

```diff
@@ -533,3 +568,38 @@ JsDebuggerUi.prototype.onStepOutButton = function() {
     this.updatePauseUiState();
   }
 };
+
+/**
+ * Refresh the watch variables.
+ * @private
+ */
+JsDebuggerUi.prototype.onWatchTimer = function() {
+  var jsInterpreter = this.jsInterpreter_;
+  if (jsInterpreter) {
+    var debugWatchDiv = this.getElement_('#debug-watch');
+    if (debugWatchDiv) {
+      while (debugWatchDiv.firstChild) {
+        debugWatchDiv.removeChild(debugWatchDiv.firstChild);
```

Comment:
You do this exact 'clear the watch area' logic up in `detach()` too.  Could extract that as a helper, but more generally I'd like to see the update and rendering logic separated here so that in `detach()` you can just say:
```javascript
this.watchVars_ = {};
this.renderWatchVars_();
```
That separation will make this easier to optimize in the future, if we decide we need to do incremental updates instead of recreating the watch div entirely, or could allow us to skip rendering if no watch values have changed.